### PR TITLE
Added two simple examples showing how integration can be run in parallel.

### PR DESCRIPTION
**Pull request by moorepants**

### CODE REVIEW

File: examples/parallel-execution/joblib_demo.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+rhs = sys.generate_ode_function(generator='cython')
+
+print('Defining wrapper functions.')
+# These two wrapper functions seem to be required for things to pickle in
+# Joblib and I'm not sure why. This is not the case if the rhs function was
+# defined as a normal Python function instead of being generated with
+# lambdify or cython.
+
+
+def rhs2(p):
```
Comment (chrisdembia) [inline examples/parallel-execution/joblib_demo.py]: `rhs_wrapper`?

File: examples/parallel-execution/joblib_demo.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from joblib import Parallel, delayed
+from pydy.models import n_link_pendulum_on_cart
+
+print('Generating EoMs')
+sys = n_link_pendulum_on_cart(10, False, False)
+
+print('Defining numerical values')
+x = np.random.random(len(sys.states))
+t = np.linspace(0.0, 10.0, 100000)
+pp = np.random.random((16, len(sys.constants_symbols)))
```
Comment (chrisdembia) [inline examples/parallel-execution/joblib_demo.py]: More meaningful name, like `params`?

File: examples/parallel-execution/joblib_demo.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+def rhs2(p):
+    return rhs(x, t[0], p)
+
+
+def odeint2(p):
+    return odeint(rhs, x, t, args=(p,))
+
+
+print('Running parallel rhs')
```
Comment (chrisdembia) [inline examples/parallel-execution/joblib_demo.py]: Move the parallel running above the def of `odeint2`? My initial impression was that these two things needed to be sequential, but it seems like you're actually just showing two distinct examples.

Comment (chrisdembia) [inline examples/parallel-execution/joblib_demo.py]: This syntax here is not obvious (the multiprocessing syntax is more obvious). Can you give a comment about what `delayed` does?

File: examples/parallel-execution/joblib_demo.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+rhs function using joblib. For example, this could be useful in genetic
+algorithm optimization routines to parallelize the evaluation of the cost
+function at each iteration."""
+
+import numpy as np
+from scipy.integrate import odeint
+from joblib import Parallel, delayed
+from pydy.models import n_link_pendulum_on_cart
+
+print('Generating EoMs')
```
Comment (chrisdembia) [inline examples/parallel-execution/joblib_demo.py]: No need to abbreviate here.

**Comment by chrisdembia** [comment]
Awesome and simple examples! I like them. Can you clarify in documentation that you are not parallelizing integration itself, but rather parallelizing the simulation across a suite of parameter sets?

I like the lead up to using pydy for design optimization.

**Comment by chrisdembia** [comment]
My review is over. Feel free to merge after addressing the comments I made.

**Comment by moorepants** [comment]
Thanks @chrisdembia I'm merging with the new changes.

**Pull request merged**